app/src/main/java/org/simple/clinic/newentry/PatientEntryScreen.kt

```diff
@@ -210,10 +210,14 @@ class PatientEntryScreen(context: Context, attrs: AttributeSet) : RelativeLayout
 
   // FIXME This is temporally coupled to `scrollToFirstFieldWithError()`.
   private val allTextInputFields: List<EditText> by unsafeLazy {
+    val ageOrDateOfBirthEditText = if (ageEditTextInputLayout.visibility == View.VISIBLE) {
+      ageEditText
+    } else {
+      dateOfBirthEditText
+    }
```

Comment:
This is not a good solution because it won't handle the case where we clear whatever field we were typing in and then start typing in the other, right?

Reply:
Just tried testing the scenario and it works for both of them. Can you try testing the same as well?

Reply:
Yup, it seems to be working. LGTM.